tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/configurables/ConfigurableTest.java

```diff
@@ -144,159 +127,126 @@ public void testInvalidTomlFile() throws BallerinaTestException {
-    @Test
-    public void testTableKeyNotFound() throws BallerinaTestException {
-        Path projectPath = Paths.get(negativeTestFileLocation, "missingTableKey").toAbsolutePath();
-        LogLeecher errorLeecher =
-                new LogLeecher("Value required for key 'username' of type " +
-                                       "'table<(main:AuthInfo & readonly)> key(username) & readonly'" +
-                                       " in configurable variable 'users'", ERROR);
-        bMainInstance.runMain("run", new String[]{"main"}, null, new String[]{},
-                              new LogLeecher[]{errorLeecher}, projectPath.toString());
-        errorLeecher.waitForText(5000);
-    }
-
-    @Test
-    public void testUnsupportedMap() throws BallerinaTestException {
-        Path projectPath = Paths.get(negativeTestFileLocation, "invalidMapType").toAbsolutePath();
-        LogLeecher errorLeecher =
-                new LogLeecher("Configurable feature is yet to be supported for type 'map<int> & readonly'", ERROR);
-        bMainInstance.runMain("run", new String[]{"main"}, null, new String[]{},
-                              new LogLeecher[]{errorLeecher}, projectPath.toString());
-        errorLeecher.waitForText(5000);
+    @DataProvider(name = "negative-tests")
+    public Object[][] getNegativeTests() {
+        return new Object[][]{
+                {"no_module_config", "Value not provided for required configurable variable 'main:stringVar'"},
+                {"invalid_org_name", "Value not provided for required configurable variable 'main:stringVar'" },
+                {"invalid_org_structure", "invalid module structure found for module 'testOrg.main'. " +
+                        "Please provide the module name as '[testOrg.main]'" },
+                {"invalid_module_structure", "invalid module structure found for module 'main'. " +
+                        "Please provide the module name as '[main]'" },
+                {"invalid_sub_module_structure", "invalid module structure found for module 'foo'. " +
+                        "Please provide the module name as '[foo]'" },
+                {"required_negative", "Value not provided for required configurable variable 'main:stringVar'"},
+                {"primitive_type_error", "invalid type found for variable 'main:intVar', expected TOML type " +
+                        "for type 'int' is 'INTEGER', found 'DOUBLE'"},
+                {"primitive_structure_error", "invalid TOML structure found for variable 'main:intVar', " +
+                        "expected structure is 'KEY_VALUE', found 'TABLE'"},
+                {"array_type_error", "invalid type found for variable 'main:intArr', expected TOML type " +
+                        "for type 'int[] & readonly' is 'ARRAY', found 'STRING'"},
+                {"array_structure_error", "invalid TOML structure found for variable 'main:intArr', " +
+                        "expected structure is 'KEY_VALUE', found 'TABLE'"},
```

Comment:
Shall we provide the error message related to the ballerina types. ex. 'intVar' is expected to be 'int' but given a record definition.